docs/rules-engine/how-to-use/custom-fact.md

```diff
@@ -0,0 +1,125 @@
+```typescript
+import type { FactDefinitions } from '@o3r/rules-engine';
+
+export interface CustomFacts extends FactDefinitions {
+  fact1: boolean;
+  fact2: string;
+}
+
+```
+* The service fact
+```typescript
+import {FactsService} from '@o3r/rules-engine';
+import type {CustomFacts} from './custom.facts.ts';
+
+@Injectable()
+export class CustomFactsService extends FactsService<CustomFacts> {
+
+  public facts = {};
+
+}
+```
+
+In order to use the fact, you need to define its logic. It can be based on manual update, on the selection of a subset
+of state, on services etc.
+
+For this example, the fact1 will be based on a store and the fact2 will be set once and will reflect the user's browser 
+agent.
+
+```typescript
+import {FactsService} from '@o3r/rules-engine';
+import {Store} from '@ngrx/store';
+import {distinctUntilChanged, filter, map, select} from 'rxjs/operators';
+import {selectUserSelection} from '../store/user-selection';
+import type {SelectionState} from '../store/user-selection';
+import type {CustomFacts} from './custom.facts.ts';
+
+@Injectable()
+export class CustomFactsService extends FactsService<CustomFacts> {
+
+  constructor(store: Store<SelectionState>) {
+    super();
+    this.facts = {
+      fact1: store.pipe(
+        filter((store) => !!store.userSelection),
+        select(selectUserSelection),
+        map((userSelection) => !!userSelection),
+        distinctUntilChanged()
+      ),
+      fact2: window.navigator.userAgent
+    }
+  }
+}
+
+```
+
+> [!INFO]
+> 
+> There are two interested thing to notice about ``fact1``:
+> * There is a pipe with a filter operator to ensure that the state is defined, before using it. This permits to register 
+ the fact even if the store is lazy loaded later in the flow. This observable will not emit until the store is loaded, but
+the rules engine will create the first emission (undefined) while waiting for the fact to emit its first value.
+> * Facts based on observable should always contain a ``distinctUntilChange()`` to avoid the re-evaluation of all the rules
+that depend on them.
+
+Then, provide the service in a dedicated module. 
+
+Note that you do not import your UserSelectionStore module here. This will allow the store to be lazy loaded when the 
+application actually requires it.
+
+```typescript
+import {RulesEngineRunnerModule} from '@o3r/rules-engine';
+import {CustomFactsService} from './custom-facts.service';
+
+@NgModule({
+  imports: [
+    RulesEngineRunnerModule
+  ],
+  providers: [
+    CustomFactsService
+  ]
+})
+export class OrderFactsModule { }
+```
+
+You can now import your fact your app module and register the facts:
+```typescript
+import {Component} from '@angular/core';
+import {CustomFactsService} from '../facts/custom-facts.service';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+})
+export class AppComponent {
+  constructor(customFactsService: CustomFactsService) {
+    customFactsService.register();
```

Comment:
Do we still want to recommand to pollute the AppComponent with the registration of facts etc...
in the `main.ts`
you could do something like 
```typescript
bootstrapApplication(AppComponent, appConfig)
   .then((m) => {
      runInInjectionContext(m.injector, () => {
        const customFactsService = inject(CustomFactsService);
        customFactsService.register();
      });
   })
```